src/evaluation/field_value.rs

```diff
@@ -0,0 +1,177 @@
+        Self::Int(value.into())
+    }
+}
+
+impl From<i16> for EvaluationContextFieldValue {
+    fn from(value: i16) -> Self {
+        Self::Int(value.into())
+    }
+}
+
+impl From<i32> for EvaluationContextFieldValue {
+    fn from(value: i32) -> Self {
+        Self::Int(value.into())
+    }
+}
+
+impl From<i64> for EvaluationContextFieldValue {
+    fn from(value: i64) -> Self {
+        Self::Int(value.into())
+    }
+}
+
+impl From<u8> for EvaluationContextFieldValue {
+    fn from(value: u8) -> Self {
+        Self::Int(value.into())
+    }
+}
+
+impl From<u16> for EvaluationContextFieldValue {
+    fn from(value: u16) -> Self {
+        Self::Int(value.into())
+    }
+}
+
+impl From<u32> for EvaluationContextFieldValue {
+    fn from(value: u32) -> Self {
+        Self::Int(value.into())
+    }
+}
+
+impl From<f32> for EvaluationContextFieldValue {
+    fn from(value: f32) -> Self {
+        Self::Float(value.into())
+    }
+}
+
+impl From<f64> for EvaluationContextFieldValue {
+    fn from(value: f64) -> Self {
+        Self::Float(value.into())
+    }
+}
+
+impl From<String> for EvaluationContextFieldValue {
+    fn from(value: String) -> Self {
+        Self::String(value)
+    }
+}
+
+impl From<&str> for EvaluationContextFieldValue {
+    fn from(value: &str) -> Self {
+        Self::String(value.to_string())
+    }
+}
+
+impl From<OffsetDateTime> for EvaluationContextFieldValue {
+    fn from(value: OffsetDateTime) -> Self {
+        Self::DateTime(value)
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::*;
+
+    #[test]
+    fn evaluation_context_custom_fields() {
+        let now = OffsetDateTime::now_utc();
+
+        let context = EvaluationContext::builder()
+            .targeting_key("Some Key")
+            .build()
+            .with_custom_field("Bool", false)
+            .with_custom_field("Bool", EvaluationContextFieldValue::Bool(true))
+            .with_custom_field("Int", 42)
+            .with_custom_field("Float", 42.0)
+            .with_custom_field("String", "StringValue")
+            .with_custom_field("DateTime", now.clone())
+            .with_custom_field(
+                "Struct",
+                EvaluationContextFieldValue::new_struct(EvaluationReason::Cached),
+            );
+
+        // Assert bool
+        if let EvaluationContextFieldValue::Bool(value) = context.custom_fields.get("Bool").unwrap()
+        {
+            assert_eq!(true, *value);
+        } else {
+            panic!()
+        };
+
+        // Assert int.
+        if let EvaluationContextFieldValue::Int(value) = context.custom_fields.get("Int").unwrap() {
+            assert_eq!(42, *value);
+        } else {
+            panic!()
+        }
+
+        // Assert float.
+        if let EvaluationContextFieldValue::Float(value) =
+            context.custom_fields.get("Float").unwrap()
+        {
+            assert_eq!(42.0, *value);
+        } else {
+            panic!()
+        }
```

Comment:
Shouldn't `unwrap` handle the panic case (here and elsewhere)?

Reply:
The `custom_fields.get()` function returns an `Option` that needs to be unwrapped. The else branch is to handle case where the returned `EvaluationContextFieldValue` is not a `Float` as desired.